### PR TITLE
fix: use FormattedPath [TFCSCM-124]

### PR DESCRIPTION
**Pull request by teodora-sandu**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?
This removes the `formattedPath` function and instead uses the `FormattedPath` field returned by `snyk-iac-test`. It also updates the version of `snyk-iac-test` to https://github.com/snyk/snyk-iac-test/releases/tag/v0.14.0.

#### Any background context you want to provide?
It contains loads of good fixes.

#### What are the relevant tickets?

- [Jira ticket TFCSCM-124](https://snyksec.atlassian.net/browse/TFCSCM-124)

### CODE REVIEW

**Comment by teodora-sandu** [comment]
Need to figure out how to generate https://github.com/snyk/snyk/blob/6f4b755b5d78b095a04a9c3fb0d57994ac6af8df/test/jest/unit/iac/process-results/fixtures/snyk-iac-test-results.json. @YairZ101 Do you have any instructions for this?

**Comment by YairZ101** [comment]
> Do you have any instructions for this?

No, when I generated this fixture I simply took several files from our goof repo, moved them to a new dir, and scanned the dir.
Do you think we should have a documented procedure to generate this fixture?

**Comment by teodora-sandu** [comment]
@YairZ101 I think documenting it would be good, even if it's just a comment at the top of this file. I assume you ran `snyk-iac-test` directly on the files to generate that output?

**Comment by YairZ101** [comment]
> @YairZ101 I think documenting it would be good, even if it's just a comment at the top of this file. I assume you ran `snyk-iac-test` directly on the files to generate that output?

I personally used the TS CLI using debug mode but you can also use `snyk-iac-test` if you prefer.

**Comment by ipapast** [comment]
hey @teodora-sandu , I found the files that Yair used from a different repo.

1. On the first commit https://github.com/snyk/cli/pull/3509/commits/fd5227f5dce7fe83cdba8088e7f417fb890ef8b1 I put them into their own fixtures folder, so we can have specific ones to test.

The json tests will not pass of course only with this commit.

2. Then, I tried to generate the expected results from this file (and trim unnessary things as the fixtures have changed quite a bit) - https://github.com/snyk/cli/pull/3509/commits/8ddd8f32969172956b4b1096718a34fec33fd2e9

3. I am now going to merge the conflicts and get the latest version for snyk-iac-test as it has bug fixes and it also involves the "float" bug you just fixed today - it's in these fixtures.

File: test/jest/unit/iac/process-results/fixtures/snyk-iac-test-results.json [hunk 1]
```diff
@@ -117,6 +117,7 @@
           "id": "aws_default_security_group.default",
           "type": "aws_default_security_group",
           "path": ["ingress", 0, "cidr_blocks", 0],
+          "formattedPath": "resource.aws_default_security_group[default].ingress[0].cidr_blocks[0]",
```
Comment (ipapast) [inline test/jest/unit/iac/process-results/fixtures/snyk-iac-test-results.json:120]: is this what you expected to see too? the "resource."

Comment (teodora-sandu) [reply test/jest/unit/iac/process-results/fixtures/snyk-iac-test-results.json:120]: Yep this is what I expected to see, but we can have a chat about it if you'd like. Thanks for taking care of the fixtures here 🙌

Comment (ipapast) [reply test/jest/unit/iac/process-results/fixtures/snyk-iac-test-results.json:120]: probably I'm not up to date hehe 😄 we should do something different with the fixtures, but for now you should be good to squash and merge.

**Review by ipapast — APPROVED** [review]

File: test/jest/unit/iac/process-results/fixtures/snyk-iac-test-results.json (hunk 1 above)
Comment (teodora-sandu) [reply test/jest/unit/iac/process-results/fixtures/snyk-iac-test-results.json:120]: Yep basically the old policy engine always added a `resource.` prefix (see https://www.notion.so/snyk/Policy-Engine-Differences-1e692f3a20ac43ae82611ae6139585a6#0ab905669fa84d7e86b46f82dc49ed3b) so we're adding it here too

**Pull request merged**